api/v1alpha1/openapi.yaml

```diff
@@ -1307,6 +1307,390 @@ paths:
+        required: true
+      responses:
+      # TODO - do we want "208" already reported, "400" bad request, "409" status conflict? Enrollment Request has these
+      # https://github.com/kubernetes/kubernetes/blob/f8d5b2074c9b3a1629ef085b9e17a5370f393732/api/openapi-spec/v3/apis__certificates.k8s.io__v1_openapi.json#L1783
+      #
+      # removed 200 as it gives an invalid reciever type error - in server.gen.go the 200JSONResponse type is interface{}
+      # enrollment request does not have this, but k8s does
+      #"200":
+      #    description: OK
+      #    content:
+      #      application/json:
+      #        schema:
+      #          ref: '#/components/schemas/CertificateSigningRequest'
+        "201":
+          description: Created
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/CertificateSigningRequest'
+        "202":
+          description: Accepted
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/CertificateSigningRequest'
+        "401":
+          description: Unauthorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+    delete:
+    # TODO: k8s has a number of parameters for this - it seems like we can add them as-needed
+    # https://github.com/kubernetes/kubernetes/blob/f8d5b2074c9b3a1629ef085b9e17a5370f393732/api/openapi-spec/v3/apis__certificates.k8s.io__v1_openapi.json#L1435
+    # Available are continue, dryRun, fieldSelector, gracePeriodSeconds, labelSelector, limit, orphanDependents, propagationPolicy,
+    # resourceVersion, resourceVersionMatch, sendInitialEvents, timeoutSeconds
+    #
+    # k8s also has a requestBody with deleteOptions - is this needed? it has properties apiVersion, dryRun, gracePeriodSeconds, kind, orphanDependents, preconditions
+    # https://github.com/kubernetes/kubernetes/blob/f8d5b2074c9b3a1629ef085b9e17a5370f393732/api/openapi-spec/v3/apis__certificates.k8s.io__v1_openapi.json#L347
+      tags:
+        - certificatesigningrequest
+      description: delete a collection of CertificateSigningRequest
+      operationId: deleteCollectionCertificateSigningRequest
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Status'
+        "401":
+          description: Unauthorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  /api/v1/certificatesigningrequests/{name}:
+    # TODO - k8s has a patch method to partially update the CSR - do we want this?
+    # https://github.com/kubernetes/kubernetes/blob/f8d5b2074c9b3a1629ef085b9e17a5370f393732/api/openapi-spec/v3/apis__certificates.k8s.io__v1_openapi.json#L2024
```

Comment:
Interesting. I was sure CSRs are immutable.

Reply:
I understood it as there are some metadata fields in the CSR resource (which contains the actual CSR in the `.spec` field - this is immutable) that do get updated, ex. with the corresponding certificate: "Commonly, the status.certificate field of a CertificateSigningRequest contains a single PEM-encoded X.509 certificate once the CSR is approved and the certificate is issued." Source: https://kubernetes.io/docs/reference/access-authn-authz/certificate-signing-requests/